topostats/io.py

```diff
@@ -193,23 +201,128 @@ def _ibw_pixel_to_nm_scaling(self, scan) -> float:
+                ch1_name = str(scan.header.data_kind_1ch).split(".")[1]
+                ch2_name = str(scan.header.data_kind_2ch).split(".")[1]
+            else:
+                raise ValueError
+        except ValueError:
+            LOGGER.error(
+                f"[{self.filename}] : File header not found. Header of type {type(scan_header)}, not [{libasd.Header_v0, libasd.Header_v1}]"
+            )
+            raise
+        try:
+            if self.channel == ch1_name:
+                channel_data = scan.channels[0]
+            elif self.channel == ch2_name:
+                channel_data = scan.channels[1]
+            else:
+                raise ValueError
+            LOGGER.info(f"[{self.filename}] : Extracted channel {self.channel} with {len(channel_data)} frames")
+            images = [channel_data[i].image() for i in range(len(channel_data))]
+        except ValueError:
+            LOGGER.error(
+                f"[{self.filename}] : {self.channel} not found in {self.suffix} channel list: [{ch1_name}, {ch2_name}"
+            )
+            raise ValueError(
+                f"[{self.filename}] : {self.channel} not found in {self.suffix} channel list: [{ch1_name}, {ch2_name}"
+            )
+
+        return (images, self._asd_px_to_nm_scaling(scan))
+
+    def _asd_px_to_nm_scaling(self, scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]) -> float:
+        """Extracts the pixel to nanometre scaling value from an .asd file object.
+
+        Parameters
+        ----------
+        scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]
+            The object resulting from loading the filepath via the libasd library.
+        Returns
+        -------
+        px_to_nm_scaling: float
+            The length of a single pixel in real length units (nm).
+
+        """
+        px_to_nm_scaling = (
+            scan.header.x_scanning_range / scan.header.x_pixel,  # scan range always in nm
+            scan.header.y_scanning_range / scan.header.y_pixel,
+        )[0]
+        return px_to_nm_scaling
+
     @staticmethod
     def _extract_jpk(jpk: AFMCreepCompliance) -> np.ndarray:
         """Extract data from jpk object"""
 
     def get_data(self) -> None:
-        """Method to extract image and pixel to nm scaling."""
-        LOGGER.info(f"Extracting image from {self.suffix}")
-        if self.suffix == ".spm":
-            self.image, self.pixel_to_nm_scaling = self.load_spm()
-        if self.suffix == ".jpk":
-            self.image, self.pixel_to_nm_scaling = self.load_jpk()
-        if self.suffix == ".ibw":
-            self.image, self.pixel_to_nm_scaling = self.load_ibw()
+        """Method to extract image, filepath and pixel to nm scaling value, and append these to the
+        img_dic object.
+        """
+        for img_path in self.img_paths:
+            self.img_path = img_path
+            self.filename = img_path.stem
+            self.suffix = img_path.suffix
```

Comment:
As with above comments about `self.suffix` because this is now within the `get_data` method and could possible change if there are files of mixed type within a given directory I think that here you can within this function do away with `self.suffix` and just use...

```suggestion
            suffix = img_path.suffix
```

(although all the `if suffix == "..."` also need updating (see above comment).

Reply:
It was also being used in the error messages I've just realised but I've changed it to collect it from `self.img_path.suffix` instead so `self.suffix `can become `suffix`